### PR TITLE
Create structured, hardcoded Home page with consistent styling

### PR DESCRIPTION
**Pull request by Mykhailo-Chudyk**
## Description

This PR develops the Home page by:
1. Adding hardcoded data to prepare for dynamic updates
2. Ensuring consistent styling across all pages
3. Enhancing code structure with clean indentation
4. Adding placeholders for future component insertions

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tests (adding or updating tests)
- [ ] Documentation update

## Screenshots (if applicable)
image

If your changes include visual components, add screenshots showing the affected pages or elements.

### CODE REVIEW

**Comment by lenovia** [comment]
I feel that the overall design is great - the color scheme and minimalistic design is great. However, I feel that it feels more designed for a mobile app. Perhaps we need to add on headers or side menus?

And also not sure if logo or name should be that big. Im also confused how big it may look when you actually insert a logo or name. What if the person also doesn't have one, what will it display?

**Comment by katyaivaniuk** [comment]
@lenovia I agree, but these are questions for our designer. I mentioned it to Saleh in the comments regarding the addition of the header as well as the display being more for a mobile version. Hence, this is something the designer needs to change asap, and then frontend engineers can develop a home page we all agree on!
But overall, the PR is great, it reflects the design that was handled to you!

**Comment by matviikotolyk** [comment]
@katyaivaniuk and @lenovia agreed! I feel like it's probably easiest to follow the traditional approach of having a full header with text for the desktop version and a toggle button that opens the menu with the tab names for mobile screens

File: src/app/home/page.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        }}
+      >
+        <Card />
+        <Card />
+      </Container>
+    </Box>
+  );
+}
+
+const Card = () => {
```
Comment (lewiskyron) [inline src/app/home/page.tsx]: This should be put into another file. In fact considering that this card has some client interactivity eg. the button inside the card we cannot have it inside the app directory. By default all the page.tsx files inside the app directory are considered server components. As a consequence we cannot implement client based interactivity on this page. 

For example if we were to implement a `ExplainMore function` that allows users to view more project info after clicking we would get the following error. 
Screenshot 2024-10-29 at 12 43 58 AM

To fix this and modularize the code further as the project grows I suggest the following. 

Create a `components` folder outside the app directory to house all client components(eg. buttons,forms e.t.c). By default, Next.js treats all components as server components, but placing them in this folder  allows us to designates them as client components. This enables client-side interactivity such as onclick & validation of user inputs. Additionally, organizing client-facing components separately keeps files lightweight, improves readability, and promotes a modular codebase—facilitating easier updates without extensive refactoring.

File: src/app/home/page.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+      <Box
+        sx={{
+          width: "100%",
+          display: "flex",
+          justifyContent: "space-around",
+          alignItems: "center",
+        }}
+      >
+        <Box sx={{ textAlign: "center", display: "flex", alignItems: "center", gap: "8px" }}>
```
Comment (lewiskyron) [inline src/app/home/page.tsx]: Create separate components for the Likes, Comments, and Views boxes. This modular approach allows each component to manage its own functionality and interactivity independently, making the code more organized and easier to maintain.

File: src/app/home/page.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+          <ModeCommentOutlinedIcon sx={{ color: "#000000", fontSize: "36px" }} />
+          <Typography sx={{ fontSize: "14px", color: "#333", fontFamily: "Outfit, sans-serif" }}>
+            {comments}
+          </Typography>
+        </Box>
+        <Box sx={{ width: "1px", backgroundColor: "#D6D6E7", height: "48px" }} />
+
+        <Box sx={{ textAlign: "center", display: "flex", alignItems: "center", gap: "8px" }}>
+          <FavoriteBorderOutlinedIcon sx={{ color: "#000000", fontSize: "36px" }} />
+          <Typography sx={{ fontSize: "14px", color: "#333", fontFamily: "Outfit, sans-serif" }}>
```
Comment (lewiskyron) [inline src/app/home/page.tsx]: unless this is specific to this boxes I assume this should be a property set in the global.css or theme provider to ensure consistency across the application.

File: src/app/home/page.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            {likes}
+          </Typography>
+        </Box>
+      </Box>
+
+      <Button
```
Comment (lewiskyron) [inline src/app/home/page.tsx]: refer to the comment made with regards to the card component.

File: src/app/home/page.tsx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+          minWidth: 120,
+          padding: "8px 16px",
+          whiteSpace: "nowrap",
+          flexShrink: 0,
+          fontFamily: "Outfit, sans-serif",
+        }}
+      >
+        Explore More
+      </Button>
+    </Box>
```
Comment (lewiskyron) [inline src/app/home/page.tsx]: I really like the final result and I think you have done a solid job in replicating the design in the figma document!

**Review by lewiskyron — CHANGES_REQUESTED** [review]

File: src/app/home/page.tsx (hunk 5 above)
Comment (lewiskyron) [reply src/app/home/page.tsx]: Finally please run npm run format to pass the formatting checks in our CI/CD pipeline.

**Comment by muhammadsaleh10** [comment]
Well-structured PR with clear, consistent styling and thoughtful use of Material UI components. I appreciate the use of the sx prop for modular, responsive styling, as well as the well-defined placeholders that make future component integration straightforward. For further improvement, I would recommend extracting repeated styles (like font family and text color) into a theme for even greater consistency. Good job!

**Review by avalucianelson — COMMENTED** [review]
This is a great PR and start for the home page, Mykhalio! Thank you so much for your contribution and explanation. This is (ofc) a critical feature for our MVP and eventual product that we'll want to perfect iteratively. 

I agree with the points team members have raised so far - Len and Katya's design concerns are valid and I think we should take that conversation elsewhere (to the Design channel in telegram, or in the 30 minutes after class today as Saleh recommended). Additionally, Kyron's feedback on specific parts of their code is critical to consider as we continue to iterate here. 

Overall, this is wonderful - you replicated the Figma designs very well and created clear and organized code. Thank you for your contribution! :)

**Comment by nnnour** [comment]
Great work, Mykhailo! The code is well-structured, and I love the attention to detail in ensuring consistent styling across pages. The placeholders for future components are a smart move and will definitely make future updates smoother. I also appreciate how well the design was translated from Figma. As others mentioned, the design feedback can be further discussed with the team, but overall, this is a great start for the Home page. I agree with Saleh and Ava that we should stay for 30 minutes after class. and if not everyone can make it, we can schedule a 30-minute meeting ASAP at a time that works for everyone. Thanks!

**Review by FlaIespa — CHANGES_REQUESTED** [review]
Good work! I also echo the comments made by the other members of team. 
I would also encourage you to use all the sections that the PR Template offers, especially the one that creates the space for providing the testing instructions, also adding the name of the branch we are meant to checkout. 
I would also suggest that you take a look at our contributions.md file, where we outline the branch naming convention we are using!

**Comment by Mykhailo-Chudyk** [comment]
> I feel that the overall design is great - the color scheme and minimalistic design is great. However, I feel that it feels more designed for a mobile app. Perhaps we need to add on headers or side menus?
> 
> And also not sure if logo or name should be that big. Im also confused how big it may look when you actually insert a logo or name. What if the person also doesn't have one, what will it display?

These are good questions, but I just followed the existing design. @muhammadsaleh10, would you please look into these comments and, if needed, update the design so that I can later create another PR to make the small changes to this page?

File: src/app/home/page.tsx (hunk 1 above)
Comment (Mykhailo-Chudyk) [reply src/app/home/page.tsx]: Agreed and updated my code!

File: src/app/home/page.tsx (hunk 2 above)
Comment (Mykhailo-Chudyk) [reply src/app/home/page.tsx]: Done!

**Comment by Mykhailo-Chudyk** [comment]
> Well-structured PR with clear, consistent styling and thoughtful use of Material UI components. I appreciate the use of the sx prop for modular, responsive styling, as well as the well-defined placeholders that make future component integration straightforward. For further improvement, I would recommend extracting repeated styles (like font family and text color) into a theme for even greater consistency. Good job!

Good comment! Agree

**Comment by Mykhailo-Chudyk** [comment]
> Good work! I also echo the comments made by the other members of team. I would also encourage you to use all the sections that the PR Template offers, especially the one that creates the space for providing the testing instructions, also adding the name of the branch we are meant to checkout. I would also suggest that you take a look at our contributions.md file, where we outline the branch naming convention we are using!

Will do. Thanks Flavia

File: src/app/home/page.tsx (hunk 5 above)
Comment (Mykhailo-Chudyk) [reply src/app/home/page.tsx]: Done

File: src/app/home/page.tsx (hunk 3 above)
Comment (Mykhailo-Chudyk) [reply src/app/home/page.tsx]: Created global variables for that

**Review by FlaIespa — APPROVED** [review]

File: src/styles/globals.css [hunk 6]
```diff
@@ -19,3 +19,24 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
+:root {
```
Comment (lewiskyron) [inline src/styles/globals.css:23]: what was in this file before. Are this styles the tailwind configs for consistent styling that other front-end devs can consume. Shouldn't block merging but is something to note.

**Review by lewiskyron — APPROVED** [review]

File: src/styles/globals.css (hunk 6 above)
Comment (Mykhailo-Chudyk) [reply src/styles/globals.css:23]: These styles establish consistent design elements (fonts, colors, spacings) across components and are intended for reuse, similar to a Tailwind configuration

**Pull request merged**